meta-client/src/main/java/org/apache/helix/metaclient/recipes/leaderelection/LeaderElectionClient.java

```diff
@@ -404,6 +416,25 @@ public void handleConnectionEstablishmentError(Throwable error) throws Exception
     }
   }
 
+  private void touchLeaderNode() {
+    for (String leaderPath : _leaderGroups) {
+      String key = leaderPath;
+      ImmutablePair<LeaderInfo, MetaClientInterface.Stat> tup = _metaClient.getDataAndStat(key);
+      if (tup.left.getLeaderName().equalsIgnoreCase(_participant)) {
+        int expectedVersion = tup.right.getVersion();
+        try {
+          _metaClient.set(key, tup.left, expectedVersion);
```

Comment:
Would this change ZNode stats? like ephemeral owner?

Also, just be safe, shall we use updater incase we have any requirement to update data in this node. At least, it will not cause race condition.

Reply:
Here when we touch current ZNode, 
1. we get ZNode version and content at the same time. 
2. Check content - if current participant is leader
3. Update ZNode using set() with expected version. Meaning we only touch ZNode if no new change was made between step 1 and 3. 

Since current participant is still leader, meaning client reconnect within expire time. Touching the ZNode wont change  ephemeral owner.

Reply:
Had an offline discussion. This will not cause data overwrite or other eace condition as we are using `set()` with expected version.